### PR TITLE
add time date dimension functions

### PR DESCRIPTION
**Pull request by tbal999**
adds a datetime function that returns a metadata object that includes time metadata from an input time

### CODE REVIEW

File: jlib/timeparse/timeparse.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	}
+
+	// convert the parsed time into a UTC time for UTC calculations
+	parsedTimeUTC := parsedTime.UTC()
+
+	dateDim := DateDim{}
+
+	// UTC TIME values
+
+	dateDim.UTC = parsedTimeUTC.Format("2006-01-02T15:04:05.000Z")
```
Comment (pocockn) [inline jlib/timeparse/timeparse.go:54]: Might be a bit clearer if you created `dateDim` via some form of constructor or maybe assigned each value to the struct in one go

```
DateDim {
    DateUTC : parsedTimeUTC.Format("2006-01-02"),
    DateID : "Dates_" + dateID
}
```

etc

File: jlib/timeparse/timeparse_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+			testObj.DateDim = actualDateDim
+			output = append(output, testObj)
+			err = json.Unmarshal(expectedByts, &expectedDateDim)
+			assert.Equal(t, expectedDateDim, actualDateDim)
+		})
+	}
+
+	outputbytes, _ := json.MarshalIndent(output, "", " ")
+	_ = os.WriteFile("outputdata.json", outputbytes, os.ModePerm)
```
Comment (pocockn) [inline jlib/timeparse/timeparse_test.go:78]: Is this for debugging purposes?

**Review by pocockn — CHANGES_REQUESTED** [review]
Few comments, overall looks good

File: jlib/timeparse/timeparse_test.go (hunk 2 above)
Comment (tbal999) [reply jlib/timeparse/timeparse_test.go:78]: yes - it's for James/Kurt to be able to easily view the input expected JSON against the output actual JSON

**Comment by tbal999** [comment]
This will remain as a PR until Kurt has reviewed & agreed that the output formats are correct.
Also we need to add more tests.

**Pull request merged**